pkg/splunk/common/urls.go

```diff
@@ -0,0 +1,134 @@
+package common
+
+// URLs for Splunk Endpoints
+// List of most common endpoints used in the Splunk Operator repo
+
+const (
+
+	//ManagerInfoAPI = "/services/cluster/master/info"
+	ManagerInfoAPI = Services + Manager + Info
+
+	//ManagerPeersAPI = "/services/cluster/master/peers"
+	ManagerPeersAPI = Services + Manager + Peers
+
+	//ClusterServiceConfig = "/services/cluster/config?"
+	ClusterServiceConfig = Services + Cluster + Config + QM
+
+	//PeerInfoAPI = "/services/cluster/slave/info"
+	PeerInfoAPI = Services + Peer + Info
+
+	//RemovePeersAPI = "/services/cluster/master/control/control/remove_peers?"
+	RemovePeersAPI = Services + Manager + Control + Control + RemovePeers + QM
+
+	//DecommissionAPI = "/services/cluster/slave/control/control/decommission?"
+	DecommissionAPI = Services + Peer + Control + Control + Decommission + QM
+
+	//ApplyBundleAPI = "/services/cluster/master/control/default/apply"
+	ApplyBundleAPI = Services + Manager + Control + Default + Apply
+
+	//SearchHeadsAPI = "/services/cluster/master/searchheads?"
+	SearchHeadsAPI = Services + Manager + SearchHeads + QM
+
+	//ManagerHealthAPI = "/services/cluster/master/health?"
+	ManagerHealthAPI = Services + Manager + Health + QM
+
+	//ManagerSitesAPI = "/services/cluster/master/sites?output_mode=json"
+	ManagerSitesAPI = Services + Manager + Sites + QM
+)
+
+// Base endpoints
+const (
+
+	//*****************
+	//   Base Hosts
+	//*****************
+
+	//Port8089 = ":8089"
+	Port8089 = ":8089"
+
+	//LocalURL = "https://localhost:8089"
+	LocalURL = "https://localhost" + Port8089
+
+	//Cluster = "/cluster"
+	Cluster = "/cluster"
+
+	//Manager = "/cluster/master"
+	Manager = Cluster + "/master"
+
+	//Peer = "/cluster/slave"
+	Peer = Cluster + "/slave"
+
+	//Spl = "splunk"
+	Spl = "splunk"
+
+	//*****************
+	//  Base Outputs
+	//*****************
+
+	//JSONOutput =  "output_mode=json"
+	JSONOutput = "output_mode=json"
+
+	//CountZero = "count=0"
+	CountZero = "count=0"
+
+	//*****************
+	//   Base APIs
+	//*****************
+
+	//Services = "/services"
+	Services = "/services"
+
+	//Info = "/info"
+	Info = "/info"
+
+	//Peers = "/peers"
+	Peers = "/peers"
+
+	//Data = "/data"
+	Data = "/data"
+
+	//Sites = "/sites"
+	Sites = "/sites"
+
+	//Health = "/health"
+	Health = "/health"
+
+	//SearchHeads = "/searchheads"
+	SearchHeads = "/searchheads"
+
+	//Indexes = "/indexes"
+	Indexes = "/indexes"
+
+	//Control = "/control"
+	Control = "/control"
+
+	//RemovePeers = "/remove_peers"
+	RemovePeers = "/remove_peers"
+
+	//Decommission = "/decommission"
+	Decommission = "/decommission"
+
+	//Default = "/default"
+	Default = "/default"
+
+	//Apply = "/apply"
+	Apply = "/apply"
+
+	//Config = "/config"
+	Config = "/config"
+
+	//LocalsPeer = "/localslave"
+	LocalsPeer = "/localslave"
+
+	//Licenser = "/licenser"
+	Licenser = "/licenser"
+
+	//QM = Question Mark
```

Comment:
Nit: Adding this comment for this entire file. If the idea behind this tokenization is to toggle the correct keyword (ex. Master vs. Manager)based on the Splunk version, then, probably, we can just limit to those terms(ex. master, slave, etc..), so that this will not be bloating going forward. 
Alternatively, If you prefer to go with this format, I think, it will be better to have some sorted order of the RHS value, so that, when somebody wants to frame a new URI/URL, they know where to look for :-)

Reply:
I agree with Subba, as much as it looks nice tokenizing the strings, it sort of is an over-kill TBH. You could have just tokenized the string in demand i.e. master vs manager or slave vs peer, etc.

Reply:
Hi @gaurav-splunk & @sgontla, this is good feedback. The main advantage of tokenization was to reduce the amount of hardcoded Paths & URLs. So in the future new constants can be defined in a single place. This is especially true for URLs that are reused many times across the code, but surely an over-kill for single usages.

But I also understand the readability concern since we no longer see a full URL in the code and the sense of an over-kill/bloating with too many tokenizations.

I think we should discuss these two options and see what the team thinks is a better approach moving forward.

I opened this sample PR for comparison: https://github.com/splunk/splunk-operator/pull/540/files

Let me know if you think that's better :D